src/components/app/check-font-loaded.js

```diff
@@ -0,0 +1,69 @@
+import { UPDATE_FONT_LOADED } from '../../actions';
```

Comment:
Does font loaded in this context mean, the font file has loaded? or the font file has been loaded and applied to the page

Reply:
Did you explore using https://github.com/typekit/webfontloader? It gives you lots of events to hook into e.g.

```
Web Font Loader provides an event system that developers can hook into. It gives you notifications of the font loading sequence in both CSS and JavaScript.

loading - This event is triggered when all fonts have been requested.
active - This event is triggered when the fonts have rendered.
inactive - This event is triggered when the browser does not support linked fonts or if none of the fonts could be loaded.
fontloading - This event is triggered once for each font that's loaded.
fontactive - This event is triggered once for each font that renders.
fontinactive - This event is triggered if the font can't be loaded.
```

The active one seems like the one you're replicating below

Reply:
> Does font loaded in this context mean, the font file has loaded? or the font file has been loaded and applied to the page

It *should* mean the latter. I think the problem is that this is an experimental feature and the implementation in each browser is pretty universally terrible.

Reply:
> Did you explore using https://github.com/typekit/webfontloader?

I didn't, because the font is added in Kedro-UI, so it would mean adding it to that package and exposing the callbacks via an API so they could be utilised in Kedro-Viz. It seemed like a lot of overhead. Seemed easier to use the built-in browser API, even if it is currently a bit inconsistent.